File: Day1/Examples/C/onePing.c
/*
 *
 * OnePing: Process 0 sends a message to process 1
 *
*/
#include <stdio.h>
#include <mpi.h>

int main(int argc, char **argv){
  int myRank, ierror, arr[5], i;    
  MPI_Status status;

  ierror=MPI_Init(&argc,&argv);
  ierror=MPI_Comm_rank(MPI_COMM_WORLD,&myRank);

  if (myRank == 0) {
    arr[0]=2345; arr[1]=654; arr[2]=96574; arr[3]=-12; arr[4]=7676;
    printf("I am %i before send ping \n", myRank);
    ierror=MPI_Send(arr, 5, MPI_INT, 1, 100, MPI_COMM_WORLD);
  } 
  else if (myRank == 1) {
    ierror=MPI_Recv(arr, 5, MPI_INT, 0, 100, MPI_COMM_WORLD, &status);
    printf("I am %i after  recv ping \n", myRank);
    for(i=0; i<5; i++) printf("arr[%d]=%d\n", i, arr[i]);
  }

  ierror=MPI_Finalize();
  return ierror;
}

File: Day1/Examples/Fortran/onePing.f90
!
! OnePing: Process 0 sends a message to process 1
!
program onePing
  use mpi_f08
  implicit none
  integer :: myRank, ierror, i
  integer :: arr(5)
  TYPE(MPI_Status) :: status

  call MPI_Init(ierror)
  call MPI_Comm_rank(MPI_COMM_WORLD,myRank,ierror)

  if (myRank .eq. 0) then
    arr=(/2345,654,96574,-12,7676/)
    write(*,*) 'I am ', myRank, ' before send ping'
    call MPI_Send(arr, 5, MPI_INTEGER, 1, 100, MPI_COMM_WORLD, ierror)
  else if (myRank .eq. 1) then
    call MPI_Recv(arr, 5, MPI_INTEGER, 0, 100, MPI_COMM_WORLD, status, ierror)
    write(*,*) 'I am ', myRank, ' after  recv ping'
    do i=1, 5
      write(*,*) 'arr(', i, ')=', arr(i)
    end do
  endif

  call MPI_Finalize(ierror)
end program onePing
